apps/microsoft-teams/frontend/src/components/config/ConfigPage/ConfigPage.spec.tsx

```diff
@@ -1,29 +1,82 @@
 import ConfigPage from './ConfigPage';
 import { describe, expect, it, vi } from 'vitest';
-import { render, screen } from '@testing-library/react';
+import { render, screen, waitFor, fireEvent } from '@testing-library/react';
 import { mockSdk } from '@test/mocks';
-import { headerSection } from '@constants/configCopy';
+import { headerSection, notificationsSection } from '@constants/configCopy';
 
 vi.mock('@contentful/react-apps-toolkit', () => ({
   useSDK: () => mockSdk,
 }));
 
+vi.mock('@azure/msal-react', () => ({
+  useMsal: vi.fn().mockReturnValue({ accounts: [{ asdf: 'asdf' }] }),
+}));
+
 vi.mock('@components/config/AccessSection/AccessSection', () => ({
   default: () => {
     return <div>Mock Access Section</div>;
   },
 }));
 
 vi.mock('@components/config/NotificationsSection/NotificationsSection', () => ({
-  default: () => {
-    return <div>Mock Notification Section</div>;
+  default: ({
+    setNotificationIndexToEdit,
+  }: {
+    setNotificationIndexToEdit: (idx: number) => void;
+  }) => {
+    return (
+      <>
+        <h1>MOCKED NOTIFICATIONS!!!!!</h1>
+        <button
+          onClick={() => {
+            setNotificationIndexToEdit(0);
+          }}
+          data-testid="edit-notification-btn">
+          edit
+        </button>
+      </>
+    );
   },
 }));
 
 describe('ConfigPage component', () => {
   it('mounts and renders title', () => {
-    render(<ConfigPage />);
+    const { unmount } = render(<ConfigPage />);
 
     expect(screen.getByText(headerSection.title)).toBeTruthy();
+
+    unmount();
+  });
+
+  describe('Saving via WidgetRenderer', () => {
+    it('prompts the User to save or cancel their pending changes when they have an open Notification modal', async () => {
+      const { unmount } = render(<ConfigPage />);
+
+      // Open an existing notification to edit it, i.e. pending change
+      let editButton;
+      await waitFor(() => {
+        // waitFor async callbacks to resolve inside useEffects
+        editButton = screen.getByTestId('edit-notification-btn');
+        return !!editButton;
```

Comment:
My suggestion above stemmed from not understanding why we need to return a boolean here. Just wasn't sure what that was needed. 😄

Reply:
To be entirely honest, I'm not sure why I was returning a boolean here, Friday was a long time ago.

I think I was attempting to placate typescript, because TS was concerned that the button was possibly undefined.

**Previous implementation**
![Screenshot 2024-02-20 at 11 44 48 AM](https://github.com/contentful/apps/assets/158083968/a8b7e610-c55e-452b-b6ce-030e8bfc0614)

I have implemented the above suggested code pattern, which is much cleaner and makes TS happy :) (which is what life is all about).